### ISSUE TITLE
Make domains cheap and elaborate an anti-parking policy

### ISSUE THREAD

**Issue by joallard** [dot-src/dot-src#3]
See ozten/TLD.js#32

**Comment by joallard** [comment]
We have to define guidelines to determine "what's an actual/admissible project"; what's not?

**Comment by nirix** [comment]
I think it goes without saying, `.src` implies source code. Obviously to be eligible the project must be open source/public domain.

**Comment by joallard** [comment]
How do we define what's a real project and what's not? How do we prevent "empty" projects? What about old, unused, defunct names?

**Comment by nirix** [comment]
We could require the project be a certain age (x weeks/months old) with active commits to it's repository, this however makes it hard for new projects to get a domain.

**Comment by joallard** [comment]
True, but we have to think of some anti-parking protection as well. We can probably find a good compromise.

Will we require a site to even have a repo? To be on Github?

**Comment by nirix** [comment]
Any publicly accessible repository should be fine.

But what about projects that only have an archive of their source and/or keep the repository out of public reach.

If we wanted to automate (even just a little) the application for a `.src`, a public repository would help with that.

**Comment by joallard** [comment]
Yep, that makes sense. Let's start drafting a "domain policy" or something like that.

**Comment by nirix** [comment]
I've been thinking about the "requirements" to apply for a `.src`
- Open source / public domain
- Publicly accessible repository
- Recent activity on the repository
- Applicant must be a member of the project team

(If we really want to stop spammers/parkers, limit to projects over a month old, it is stupid though)

The application process can probably be automated a little by checking the repository commit history for activity and estimated project age. If the applicants name/email is in the repository history this will help, even if just a little, with verifying the applicant has something to do with the project.

**Comment by arxpoetica** [comment]
That whole month thing feels like it'd kill it dead in the water. Is there a better idea?

**Comment by nirix** [comment]
Could make it a week?

**Comment by joallard** [comment]
I don't even think we need the time limit in the first place. I like the idea of being able to get a project up and running in 24 hours.

**Comment by nirix** [comment]
True, a public repository and the applicant being part of the project should be enough.

**Comment by cedricdonie** [comment]
I guess that we just need a checklist with all criteria that something needs to fulfill.
Here is a basic idea. (I will create a more rigorous list of criteria later on)
- source code must be licensed under one of the OSI-approved licenses (http://opensource.org/licenses/alphabetical)
- Project must be in active development OR still be highly frequented and products / code still downloaded (We would't take down the `python.py.src` domain even if python development would stop, as the current code and binaries are still of immense value)
- Reselling registered domains is not allowed (except for the price of one month rental to compensate for the money lost if domain is sold before end of month). Sample scenario:
  
  > Company Parker & Sons registers `foobar.js.src`. They workaround the ctirteria (e.g. fork from another project and change a few lines of code in 10 minute) to be allowed to register this domain, because they believe that `foobbar.js.src` is of extremely large value, so the main goal is selling it to a company. Then Mr. McUnix comes along and contacts Parker & Sons, asking them it they could free up the domain, as he is developing an important open-source project. Of course, Parker & Sons will not just free up the domain, but they send him a quote of 1000USD for the domain rights. No Mr. McUnix is thrilled (really). All that he needs to do is to send the evidence of Parker & Sons trying to sell him the domain. The people in charge of the `.src` domain then  end his contract and Mr- McUnix can now register without paying a penny to Parker & Sons. 
  > Of course, we would then need to include a clause that whoever responsible can end the contract anytime at their jurisdiction.
- People applying need to first create a public repo, and be part of it. There is a one-week barrier after starting the repo for (github, etc.) users who don't have any previous projects. Otherwise I would say it takes 1-2 days.
- Any project that is partially open-source (e.g. Qt professional and open-source), only the open-source portion may be on the .src domain. While it is okay to include a link on the `.src`-domain website, the target content must be hosted on a different domain.

I will come up with more later...

**Comment by joallard** [comment]
I like it! We could make this into a delegation policy file in the repo!

**Comment by cedricdonie** [comment]
Few more points:
- Although obvious, `.src` domains are only intended for use with open source software. This means that a website delivering free, GPL stock photos may not use a `.src` domain. 
- <s>No domain redirects to another non-.src domain. (maybe we can allow 403 redirects if the site has been moved, and if the redirect goes to the new project site)</s>
- Absolutely no typo squatting  If I type in `jqeury.js.src`, I expect to get an error, or get redirected to the JavaScript framework, not other sites that *ahem...* have very little to do with code.

**Comment by mhelvens** [comment]
To be honest with you, I find most of these rules rather arbitrary. Let's look at the latest three:

> Although obvious, .src domains are only intended for use with open source software. This means that a website delivering free, GPL stock photos may not use a .src domain.

To be honest, I don't find it all that obvious, and it all depends on your definition of 'software'. When 'software' is everything that is not 'hardware', photos certainly qualify. If it has to be a 'computer program' (which is 'executable'), you throw out HTML and CSS code. What about SVG, PostScript, or LaTeX? LaTeX is a language to describe technical documents, but it happens to be Turing complete.

> No domain redirects to another non-.src domain. (maybe we can allow 403 redirects if the site has been moved, and if the redirectgoes to the new project site:)

With this one rule, you exclude all pre-existing open source projects from getting a `.src` domain.

> Absolutely no typo squatting If I type in jqeury.js.src, I expect to get an error, or get redirected to the JavaScript framework, not other sites that _ahem..._ have very little to do with code.

The problem with these kinds of rules is that they are impossible to judge. When is something a 'typo squat' rather than a legitimate project with a similar name? Will you enforce a Levenshtein distance between existing domain-names? Playing the 'common sense' card will get you into all kinds of (legal) trouble. Rather, domain-name squatters will probably not be hosting legitimate and active open source projects, and they can be excluded on those grounds.

**As a general point:** Be careful of creating rules just because you can.

**Comment by cedricdonie** [comment]
@mhelvens I see your point, I should have thought the rules out a little more. Now that I think of it, the anti-redirect does seem pointless.
We will have to agree on what is illegible "software", and what is not okay for the domain.
